### PR TITLE
Add /api/schemas endpoint to vtadmin-api

### PR DESCRIPTION
**Pull request by doeg**
Signed-off-by: Sara Bee <855595+doeg@users.noreply.github.com>

## Description

A simplified version of our Slack-internal /api/schemas endpoint. Notably:
- This does not provide a mechanism to filter by anything other than cluster ID
- This does not implement any caching

"Don't forget" to expand the `api_test.go` diff 😭 

@ajm188 thanks as always for weathering my ~~incessant whining~~ questions! 

### Queries against local Vitess

<details>
<summary><code>curl "http://localhost:14200/api/schemas" | jq .</code></summary>

```json
{
  "result": {
    "schemas": [
      {
        "cluster": {
          "id": "id1",
          "name": "cluster1"
        },
        "keyspace": "commerce",
        "table_definitions": [
          {
            "name": "customer_seq",
            "schema": "CREATE TABLE `customer_seq` (\n  `id` int(11) NOT NULL,\n  `next_id` bigint(20) DEFAULT NULL,\n  `cache` bigint(20) DEFAULT NULL,\n  PRIMARY KEY (`id`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8 COMMENT='vitess_sequence'",
            "columns": [
              "id",
              "next_id",
              "cache"
            ],
            "primary_key_columns": [
              "id"
            ],
            "type": "BASE TABLE",
            "data_length": 16384,
            "row_count": 1,
            "fields": [
              {
                "name": "id",
                "type": 263,
                "table": "customer_seq",
                "org_table": "customer_seq",
                "database": "vt_commerce",
                "org_name": "id",
                "column_length": 11,
                "charset": 63,
                "flags": 53251
              },
              {
                "name": "next_id",
                "type": 265,
                "table": "customer_seq",
                "org_table": "customer_seq",
                "database": "vt_commerce",
                "org_name": "next_id",
                "column_length": 20,
                "charset": 63,
                "flags": 32768
              },
              {
                "name": "cache",
                "type": 265,
                "table": "customer_seq",
                "org_table": "customer_seq",
                "database": "vt_commerce",
                "org_name": "cache",
                "column_length": 20,
                "charset": 63,
                "flags": 32768
              }
            ]
          },
          {
            "name": "order_seq",
            "schema": "CREATE TABLE `order_seq` (\n  `id` int(11) NOT NULL,\n  `next_id` bigint(20) DEFAULT NULL,\n  `cache` bigint(20) DEFAULT NULL,\n  PRIMARY KEY (`id`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8 COMMENT='vitess_sequence'",
            "columns": [
              "id",
              "next_id",
              "cache"
            ],
            "primary_key_columns": [
              "id"
            ],
            "type": "BASE TABLE",
            "data_length": 16384,
            "row_count": 1,
            "fields": [
              {
                "name": "id",
                "type": 263,
                "table": "order_seq",
                "org_table": "order_seq",
                "database": "vt_commerce",
                "org_name": "id",
                "column_length": 11,
                "charset": 63,
                "flags": 53251
              },
              {
                "name": "next_id",
                "type": 265,
                "table": "order_seq",
                "org_table": "order_seq",
                "database": "vt_commerce",
                "org_name": "next_id",
                "column_length": 20,
                "charset": 63,
                "flags": 32768
              },
              {
                "name": "cache",
                "type": 265,
                "table": "order_seq",
                "org_table": "order_seq",
                "database": "vt_commerce",
                "org_name": "cache",
                "column_length": 20,
                "charset": 63,
                "flags": 32768
              }
            ]
          },
          {
            "name": "product",
            "schema": "CREATE TABLE `product` (\n  `sku` varbinary(128) NOT NULL,\n  `description` varbinary(128) DEFAULT NULL,\n  `price` bigint(20) DEFAULT NULL,\n  PRIMARY KEY (`sku`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8",
            "columns": [
              "sku",
              "description",
              "price"
            ],
            "primary_key_columns": [
              "sku"
            ],
            "type": "BASE TABLE",
            "data_length": 16384,
            "fields": [
              {
                "name": "sku",
                "type": 10262,
                "table": "product",
                "org_table": "product",
                "database": "vt_commerce",
                "org_name": "sku",
                "column_length": 128,
                "charset": 63,
                "flags": 20611
              },
              {
                "name": "description",
                "type": 10262,
                "table": "product",
                "org_table": "product",
                "database": "vt_commerce",
                "org_name": "description",
                "column_length": 128,
                "charset": 63,
                "flags": 128
              },
              {
                "name": "price",
                "type": 265,
                "table": "product",
                "org_table": "product",
                "database": "vt_commerce",
                "org_name": "price",
                "column_length": 20,
                "charset": 63,
                "flags": 32768
              }
            ]
          }
        ]
      }
    ]
  },
  "ok": true
}
```
</details>

<details>
<summary><code>curl "http://localhost:14200/api/schemas?cluster=nope" | jq .</code></summary>

```json
{
  "result": {},
  "ok": true
}
```
</details>

## Related Issue(s)

- https://github.com/vitessio/vitess/pull/7346

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

## Deployment Notes

N/A

## Impacted Areas in Vitess
Components that this PR will affect:

- [ ]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build/CI
- [x]  VTAdmin

### CODE REVIEW

File: go/vt/vtadmin/api.go [hunk 1]
```diff
@@ -242,6 +243,154 @@ func (api *API) GetKeyspaces(ctx context.Context, req *vtadminpb.GetKeyspacesReq
...
+		return nil, er.Error()
+	}
+
+	return &vtadminpb.GetSchemasResponse{
+		Schemas: schemas,
+	}, nil
+}
+
+// getSchemas returns all of the schemas across all keyspaces in the given cluster.
+func (api *API) getSchemas(ctx context.Context, c *cluster.Cluster, tablets []*vtadminpb.Tablet) ([]*vtadminpb.Schema, error) {
```
Comment (ajm188) [inline go/vt/vtadmin/api.go:299]: TIOLI: I know we talked a bit about this file potentially growing very large as the API surface area grows, and these methods don't actually depend on `api` for anything, so if you wanted to you could make these top-level private functions in another file, but also this is fine as-is.

File: go/vt/vtadmin/api_test.go [hunk 2]
```diff
@@ -41,10 +43,13 @@ import (
 	"vitess.io/vitess/go/vt/vtctl/vtctldclient"
 	"vitess.io/vitess/go/vt/vttablet/tmclient"
 
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	"vitess.io/vitess/go/vt/proto/tabletmanagerdata"
 	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 	vtadminpb "vitess.io/vitess/go/vt/proto/vtadmin"
 	vtctldatapb "vitess.io/vitess/go/vt/proto/vtctldata"
 	"vitess.io/vitess/go/vt/proto/vttime"
+	vtadmintestutil "vitess.io/vitess/go/vt/vtadmin/testutil"
```
Comment (ajm188) [inline go/vt/vtadmin/api_test.go:52]: sadly vscode doesn't always do this right, but for consistency this should be up with the non-proto vitess imports

File: go/vt/vtadmin/api_test.go [hunk 3]
```diff
@@ -271,6 +276,481 @@ func TestGetKeyspaces(t *testing.T) {
...
 
+func TestGetSchemas(t *testing.T) {
+	tests := []struct {
+		name           string
+		clusterTablets [][]*vtadminpb.Tablet
+		// Indexed by tablet alias
+		tabletSchemas map[string]*tabletmanagerdata.SchemaDefinition
+		req           *vtadminpb.GetSchemasRequest
+		expected      *vtadminpb.GetSchemasResponse
+	}{
```
Comment (ajm188) [inline go/vt/vtadmin/api_test.go:287]: Ugh, I am _sorry_ lol. These are great tests though!

**Review by ajm188 — APPROVED** [review]
lg2m!

File: go/vt/vtadmin/api.go (hunk 1 above)
Comment (doeg) [reply go/vt/vtadmin/api.go:299]: I'll leave it for now. I agree with you on one file offering better readability. Even though these methods don't use `api` directly, the colocation seems good. One day I'll get used to the verbosity.

File: go/vt/vtadmin/api_test.go (hunk 2 above)
Comment (doeg) [reply go/vt/vtadmin/api_test.go]: ✅

**Review by ajm188 — APPROVED** [review]

**Pull request merged**